includes/Checker/Checks/Enqueued_Scripts_Size_Check.php

```diff
@@ -0,0 +1,229 @@
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	private $threshold_size;
+
+	/**
+	 * List of viewable post types.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $viewable_post_types;
+
+	/**
+	 * Set the threshold size for script sizes to surface warnings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param integer $threshold_size The threshold in bytes for script size to surface warnings.
+	 */
+	public function __construct( $threshold_size = 300000 ) {
+		$this->threshold_size = $threshold_size;
+
+		$this->viewable_post_types = array_filter(
+			get_post_types(),
+			function( $post_type ) {
+				return is_post_type_viewable( $post_type );
+			}
+		);
+	}
+
+	/**
+	 * Runs this preparation step for the environment and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		$orig_scripts = isset( $GLOBALS['wp_scripts'] ) ? $GLOBALS['wp_scripts'] : null;
+
+		// Backup the original values for the global state.
+		$check = $this;
+		$check->backup_globals();
+
+		return function() use ( $orig_scripts, $check ) {
+			if ( is_null( $orig_scripts ) ) {
+				unset( $GLOBALS['wp_scripts'] );
+			} else {
+				$GLOBALS['wp_scripts'] = $orig_scripts;
+			}
+
+			$check->restore_globals();
+		};
+	}
+
+	/**
+	 * Returns an array of shared preparations for the check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->viewable_post_types
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+
+	/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+		$this->run_for_urls(
+			$this->get_urls(),
+			function ( $url ) use ( $result ) {
+				$this->check_url( $result, $url );
+			}
+		);
+	}
+
+	/**
+	 * Gets the list of URLs to run this check for.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of URL strings (either full URLs or paths).
+	 */
+	protected function get_urls() {
+		$urls = array( home_url() );
+
+		foreach ( $this->viewable_post_types as $post_type ) {
+			$posts = get_posts(
+				array(
+					'posts_per_page' => 1,
+					'post_type'      => $post_type,
+					'post_status'    => 'publish',
+				)
+			);
+			if ( isset( $posts[0] ) ) {
+				$urls[] = get_permalink( $posts[0] );
+			}
```

Comment:
Rather than silently only adding the URL if the post is present in the result, we should throw an exception if a post is missing. If the post is not returned, something went wrong, and our code should fail, rather than the check not being reliable for an "unknown" reason.